### PR TITLE
Updates to SDK to support .NET Framework targets

### PR DESCRIPTION
**Pull request by fabiocav**
### Pull request checklist

* [x] My changes **do not** require documentation changes
  * [ ] Otherwise: Documentation issue linked to PR
* [x] My changes **should not** be added to the release notes for the next release
  * [ ] Otherwise: I've added my notes to `release_notes.md`
* [x] My changes **do not** need to be backported to a previous version
  * [ ] Otherwise: Backport tracked by issue/PR #issue_or_pr
* [x] I have added all required tests (Unit tests, E2E tests)

### Additional information

Additional PR information

### CODE REVIEW

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 1]
```diff
@@ -11,211 +11,239 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
+          AzureFunctionsVersion="$(AzureFunctionsVersion)"
+          OutputPath="$(TargetDir)"/>
+
+        <Copy
+          SourceFiles="$(_FunctionsMetadataLoaderExtensionFile)"
+          DestinationFolder="$(ExtensionsCsProjFilePath)\buildout"
+          OverwriteReadOnlyFiles="true" />
+
+        <WriteLinesToFile
+            Condition="$(TargetFrameworkIdentifier) == '.NETFramework'"
```
Comment (kshyju) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:79]: Should we remove the `Condition` expression as it is already in the parent level (line 60)?

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 2]
```diff
@@ -225,122 +253,151 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
+
+        <WriteLinesToFile
+            File="$(OutputFile)"
+            Lines="$([System.IO.File]::ReadAllText($(_FunctionsWorkerConfigInputFile))
+                                  .Replace('$functionExe$', '{WorkerRoot}$(TargetName).exe')
+                                  .Replace('$functionWorker$', '$(TargetName).exe'))"
+            Overwrite="true" />
+    </Target>
+
+    <Target Name="_GenerateFunctionsAndCopyContentFilesNetApp" Condition="$(TargetFrameworkIdentifier) == '.NETFramework'">
```
Comment (kshyju) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:293]: Question: Should the file extensions used in the `WriteLinesToFile` below (312,320) be fixed from .dll to .exe?

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets [hunk 3]
```diff
@@ -11,211 +11,239 @@ WARNING:  DO NOT MODIFY this file unless you are knowledgeable about MSBuild and
...
-      <ExtensionsCsProjFilePath>$([System.IO.Path]::Combine($([System.IO.Path]::GetTempPath()), $([System.IO.Path]::GetRandomFileName())))</ExtensionsCsProjFilePath>
+        <_ToolingSuffix></_ToolingSuffix>
+        <_AzureFunctionsNotSet Condition="'$(AzureFunctionsVersion)' == ''">true</_AzureFunctionsNotSet>
+        <AzureFunctionsVersion Condition="'$(AzureFunctionsVersion)' == ''">v3</AzureFunctionsVersion>
+        <_ToolingSuffix Condition="($(AzureFunctionsVersion.StartsWith('v3',StringComparison.OrdinalIgnoreCase)) Or $(AzureFunctionsVersion.StartsWith('v4',StringComparison.OrdinalIgnoreCase))) And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == 'v5.0'">net5-isolated</_ToolingSuffix>
+        <_ToolingSuffix Condition="$(AzureFunctionsVersion.StartsWith('v4',StringComparison.OrdinalIgnoreCase)) And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' And '$(TargetFrameworkVersion)' == 'v6.0'">net6-isolated</_ToolingSuffix>
+        <_ToolingSuffix Condition="$(AzureFunctionsVersion.StartsWith('v4',StringComparison.OrdinalIgnoreCase)) And '$(TargetFrameworkIdentifier)' == '.NETFramework'">net6-isolated</_ToolingSuffix>
```
Comment (kshyju) [inline sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:20]: Should we enforce a minimum version of .NET framework as the framework value / Can one use .NET framework 3.0 ?

Comment (fabiocav) [reply sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:20]: The version here will be enforced by the targets. Anything that supports .NET Standard 2.0 would be fine.

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets (hunk 1 above)
Comment (fabiocav) [reply sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:79]: 👍🏼

File: sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets (hunk 2 above)
Comment (fabiocav) [reply sdk/Sdk/Targets/Microsoft.Azure.Functions.Worker.Sdk.targets:293]: No, but your comment made me realize that the condition here was incorrectly checking for equality.

**Pull request merged**